storm-webapp/src/main/java/org/apache/storm/daemon/ui/UIHelpers.java

```diff
@@ -1519,8 +1518,12 @@ private static Double nullToZero(Double value) {
         result.put("executeLatency", StatsUtil.floatStr(boltAggregateStats.get_execute_latency_ms()));
         result.put("executed", boltAggregateStats.get_executed());
         result.put("processLatency", StatsUtil.floatStr(boltAggregateStats.get_process_latency_ms()));
+
         ErrorInfo lastError = componentAggregateStats.get_last_error();
         result.put("lastError", Objects.isNull(lastError) ?  "" : getTruncatedErrorString(lastError.get_error()));
+        if (!Objects.isNull(lastError)) {
```

Comment:
The existing [getComponentErrorInfo()](https://github.com/Crim/storm/blob/sp/STORM-3652/storm-webapp/src/main/java/org/apache/storm/daemon/ui/UIHelpers.java#L1430-L1443) method adds the remaining properties that the UI is expecting to properly render the error.  See [topology-page-template.html](https://github.com/Crim/storm/blob/sp/STORM-3652/storm-webapp/src/main/java/org/apache/storm/daemon/ui/WEB-INF/templates/topology-page-template.html#L565-L574)

Reply:
We can add a parameter to `getComponentErrorInfo(xxx, boolean asLastError)` and when it is lastError, 
```
if (asLastError) {
result.put("lastError",getTruncatedErrorString)errorInfo.get_error()));  
} else {
result.put("error", errorInfo.get_error());
}
```
 to avoid unnecessary results